### PR TITLE
[ML] Remove top level importance from classification inference results

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
As we have decided top level importance for classification is not useful,
it has been removed from the results from the training job. This commit
also removes them from inference.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/FeatureImportance.java [hunk 1]
```diff
@@ -104,22 +109,28 @@ public String getFeatureName() {
...
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeString(this.featureName);
-        out.writeDouble(this.importance);
-        out.writeBoolean(this.classImportance != null);
-        if (this.classImportance != null) {
+        out.writeString(featureName);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeOptionalDouble(importance);
+        } else {
+            out.writeDouble(importance);
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/FeatureImportance.java:116]: couldn't this blow up? we need to somehow require that `importance` here is not null.

I think the way to do that is to run 
```
            classImportance.stream().mapToDouble(ClassImportance::getImportance).map(Math::abs).sum()
```
if it is null. If classImportance is null && importance is null, then maybe just return a 0.0

Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/FeatureImportance.java:116]: Indeed!

**Review by benwtrent — APPROVED** [review]

**Comment by dimitris-athanasiou** [comment]
@benwtrent I have now split feature importance into two separate classes, one per analysis type. Could you please take another look?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationFeatureImportance.java [hunk 2]
```diff
@@ -26,157 +26,102 @@
...
-            } else {
-                this.classImportance = in.readList(ClassImportance::new);
-            }
-        } else {
-            this.classImportance = null;
-        }
+        this.classImportance = in.readList(ClassImportance::new);
     }
 
+    @Nullable
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationFeatureImportance.java:64]: I don't think this is nullable.
```suggestion
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationFeatureImportance.java [hunk 3]
```diff
@@ -5,7 +5,7 @@
  */
 package org.elasticsearch.xpack.core.ml.inference.results;
 
-import org.elasticsearch.Version;
+import org.elasticsearch.common.Nullable;
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationFeatureImportance.java:8]:
```suggestion
```

**Review by benwtrent — APPROVED** [review]

**Pull request merged**